tests/algorithms/test_torchscript.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import torch
+from torchvision.models import resnet50
+
+from composer.functional import (apply_blurpool, apply_factorization, apply_ghost_batchnorm, apply_squeeze_excite,
+                                 apply_stochastic_depth)
+
+algo_kwargs = {apply_stochastic_depth: {'stochastic_method': 'block', 'target_layer_name': 'ResNetBottleneck'}}
+
+
+@pytest.fixture
+def input():
+    return torch.Tensor(1, 3, 224, 224)
+
+
+@pytest.mark.parametrize("surgery_method", [
+    pytest.param(apply_blurpool),
+    pytest.param(apply_factorization, marks=pytest.mark.xfail),
+    pytest.param(apply_ghost_batchnorm, marks=pytest.mark.xfail),
+    pytest.param(apply_squeeze_excite),
+    pytest.param(apply_stochastic_depth, marks=pytest.mark.xfail),
+])
+def test_surgery_torchscript_train(surgery_method, input):
+    """Tests torchscript model in train mode."""
+    model = resnet50()
+    kwargs = algo_kwargs.get(surgery_method, {})
+
+    model = surgery_method(model, **kwargs)
+    model.train()
+
+    scripted_func = torch.jit.script(model)
+    scripted_func(input)  # type: ignore (third-party)
+
+
+@pytest.mark.parametrize("surgery_method", [
+    pytest.param(apply_blurpool),
+    pytest.param(apply_factorization, marks=pytest.mark.xfail),
+    pytest.param(apply_ghost_batchnorm, marks=pytest.mark.xfail),
+    pytest.param(apply_squeeze_excite),
+    pytest.param(apply_stochastic_depth),
+])
+def test_surgery_torchscript_eval(surgery_method, input):
```

Comment:
some duplicated code here for eval so reader can easily tell which algorithms are TS-supportable in what modes from the `parameterize` list.